metrics.go

```diff
@@ -11,27 +11,35 @@ import (
 // with values relative to that last second.
 type MetricsBag map[string]interface{}
 
-// AddRawCounter adds a counter that always increments.
+// AddRawCounter adds a counter. A counter is a cumulative metrics that can only increase.
+// value must be positive.
```

Comment:
it seems we enforce a check, but we don't, so maybe I'd write:

```suggestion
// To be meaningful, `value` must be positive.
```